packages/react/src/search/Search.tsx

```diff
@@ -0,0 +1,54 @@
+import { type ComponentPropsWithRef, forwardRef, useState } from "react";
+
+import { Box } from "../box";
+import { Button } from "../button";
+import { Input } from "../input";
+import { IconCross } from "./IconCross";
+import { IconSearch } from "./IconSearch";
+
+type SearchProps = ComponentPropsWithRef<typeof Input>;
+
+export const Search = forwardRef<HTMLInputElement, SearchProps>(
+  ({ defaultValue = "", onChange, ...props }, ref) => {
+    const [inputValue, setInputValue] = useState(defaultValue);
+
+    const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setInputValue(event.target.value);
+      if (onChange) {
+        onChange(event);
+      }
+    };
+
+    const handleClearInput = () => {
+      setInputValue("");
+      if (onChange) {
+        onChange({
+          target: { value: "" },
+        } as React.ChangeEvent<HTMLInputElement>);
```

Comment:
this would be problematic... somebody's gonna try to access something other than `.target.value` and face runtime error.

Reply:
Yes vai, I also thought that.
Should I add a onClear prop rather than using the onChange?